cmd/stack_lint_test.go

```diff
@@ -93,6 +90,59 @@ func TestLinterInvalidValues(t *testing.T) {
 		})
 	}
 }
+
+func TestLinterWarnings(t *testing.T) {
+	var dockerfilePath = filepath.Join("image", "Dockerfile-stack")
+
+	var linterInvalidValues = []struct {
+		testName     string
+		targetPath   string
+		containsLine string
+		replaceLine  string
+		expectedLog  string
+	}{
+		{"Invalid Run Value", dockerfilePath, "APPSODY_RUN", "Testing", "Missing APPSODY_RUN"},
+		{"Invalid Watch Dir", dockerfilePath, "_ON_CHANGE", "Testing", "APPSODY_WATCH_DIR is defined, but no ON_CHANGE variable is defined"},
+		// Fails unmarshalling the file when searching for "name" - searching for "sample stack" instead
+		{"Invalid License Field", "stack.yaml", "license: ", "license: invalidLicense", "The stack.yaml SPDX license ID is invalid"},
+	}
+	for _, testData := range linterInvalidValues {
+		// need to set testData to a new variable scoped under the for loop
+		// otherwise tests run in parallel may get the wrong testData
+		// because the for loop reassigns it before the func runs
+		tt := testData
+		// call t.Run so that we can name and report on individual tests
+		t.Run(tt.testName, func(t *testing.T) {
+			sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
+			defer cleanup()
+			testStackPath := filepath.Join(sandbox.TestDataPath, "test-stack")
+			targetPath := filepath.Join(testStackPath, tt.targetPath)
+			file, err := ioutil.ReadFile(targetPath)
+			if err != nil {
+				t.Fatal(err)
+			}
+			lines := strings.Split(string(file), "\n")
+			for i, line := range lines {
+				if strings.Contains(line, tt.containsLine) {
+					lines[i] = tt.replaceLine
+				}
+			}
+			output := strings.Join(lines, "\n")
+			err = ioutil.WriteFile(targetPath, []byte(output), 0644)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
I see a lot of common code with the other tests. This could be put into a function in the future